open-api/rest-catalog-open-api.yaml

```diff
@@ -222,7 +222,8 @@ paths:
           description:
             An optional namespace, underneath which to list namespaces.
             If not provided or empty, all top-level namespaces should be listed.
-            If parent is a multipart namespace, the parts must be separated by the unit separator (`0x1F`) byte.
+            If parent is a multipart namespace, the parts must be separated by the namespace separator as
+            indicated via the /config override `rest-namespace-separator`, which defaults to the unit separator (`0x1F`) byte.
```

Comment:
This approach will break old client in environments where the server is able to accept the non-printable separator, but chooses to use an alternative separator.

So, the upgrade path for users would be to first upgrade clients, then use new servers. As for me, it can be a significant burden on users. Also, this will require that new clients be API-compatible with older servers during the transition period... WDYT?

Reply:
This won't break old clients and there's even a test that makes sure old clients send `%1F` while the server chose `%2E`. See `TestRestUtil#encodeAsOldClientAndDecodeAsNewServer()`

Reply:
Please add that to the spec. I do not think we can assume that all REST server implementation reuse Iceberg _java_ code.

Reply:
As it stands now "must" in line 226 disallows clients to use the old separator.